test/e2e/bundle_test.go

```diff
@@ -0,0 +1,137 @@
+	} else {
+		podName, err = data.getAntreaPodOnNode(masterNodeName())
+		podPort = fmt.Sprint(apis.AntreaAgentAPIPort)
+		tokenPath = agentapiserver.TokenPath
+	}
+	require.NoError(t, err)
+	// acquired token.
+	token, err := getAccessToken(podName, fmt.Sprintf("antrea-%s", name), tokenPath, data)
+	require.NoError(t, err)
+	podIP, err := data.podWaitForIP(defaultTimeout, podName, metav1.NamespaceSystem)
+	require.NoError(t, err)
+	// setup clients.
+	localConfig := rest.CopyConfig(data.kubeConfig)
+	localConfig.Host = net.JoinHostPort(podIP, podPort)
+	localConfig.BearerToken = token
+	localConfig.Insecure = true
+	localConfig.CAFile = ""
+	localConfig.CAData = nil
+	clients, err := clientset.NewForConfig(localConfig)
+	require.NoError(t, err)
+	// Clearing any exists bundle.
+	err = clients.SystemV1beta1().Bundles().Delete(name, &metav1.DeleteOptions{})
+	require.NoError(t, nil)
+	time.Sleep(100 * time.Millisecond)
+	// Checking the initial status.
+	bundle, err := clients.SystemV1beta1().Bundles().Get(name, metav1.GetOptions{})
+	require.NoError(t, err)
+	require.Equal(t, systemv1beta1.BundleNone, bundle.Status)
+	// Creating a new bundle.
+	bundle, err = clients.SystemV1beta1().Bundles().Create(&systemv1beta1.Bundle{ObjectMeta: metav1.ObjectMeta{Name: name}})
+	require.NoError(t, err)
+	require.Equal(t, systemv1beta1.BundleCollecting, bundle.Status)
+	// Waiting the generation to be completed.
+	ddl := time.After(defaultTimeout)
+	err = wait.PollImmediateUntil(200*time.Millisecond, func() (done bool, err error) {
+		select {
+		case <-ddl:
+			return false, fmt.Errorf("collecting timeout")
+		default:
+		}
+		bundle, err = clients.SystemV1beta1().Bundles().Get(name, metav1.GetOptions{})
+		require.NoError(t, err)
+		return bundle.Status == systemv1beta1.BundleCollected, nil
+	}, nil)
+	require.NoError(t, err)
+	// Checking the complete status.
+	bundle, err = clients.SystemV1beta1().Bundles().Get(name, metav1.GetOptions{})
+	require.NoError(t, err)
+	require.Equal(t, systemv1beta1.BundleCollected, bundle.Status)
+	// Downloading the bundle and verify sha256sum.
+	readStream, err := clients.SystemV1beta1().RESTClient().
+		Get().
+		Resource("bundles").
+		Name(name).
+		SubResource("download").
+		Stream()
+	require.NoError(t, err)
+	defer readStream.Close()
+	hasher := sha256.New()
+	_, err = io.Copy(hasher, readStream)
+	require.NoError(t, err)
+	require.Equal(t, bundle.Sum, fmt.Sprintf("%x", hasher.Sum(nil)))
+	// Deleting the bundle.
+	err = clients.SystemV1beta1().Bundles().Delete(name, &metav1.DeleteOptions{})
+	require.NoError(t, nil)
+	time.Sleep(100 * time.Millisecond)
+	// Checking if the bundle deleted.
+	bundle, err = clients.SystemV1beta1().Bundles().Get(name, metav1.GetOptions{})
+	require.NoError(t, err)
+	require.Equal(t, systemv1beta1.BundleNone, bundle.Status)
+}
+
+func TestBundleController(t *testing.T) {
+	testBundle("controller", t)
+}
+
+func TestBundleAgent(t *testing.T) {
+	testBundle("agent", t)
```

Comment:
could you confirm that as far as the API is concerned, these names can be arbitrary? For example, can I use "foo" / "bar" instead of "agent" / "controller" when creating / getting the bundle resource?

with that in mind I should be able to create multiple bundles "concurrently" for a given agent, by using different names?

Reply:
The storage only accepts resources that have same name of its mode, thus other names would not work. I could add a test about it.

Reply:
I am a bit confused at to why we have this restriction. Could you clarify why we cannot support arbitrary resource names for the bundles? I think it would make the API more similar to other K8s resource APIs. It also means that we could enable multiple independent clients to create separate bundles "in parallel" using different names (I don't know how useful this feature would be...). What do you think?

Reply:
Hi, Antonin, Weiqiang. For simplicity I think as a start it can support only 1 bundle at a time. To support parallel collection, it has to store more files at server side. I'm not sure if people want to collect generate multiple support bundles at the same time, because the collected results would be almost identical. So maybe 1 bundle at time would be sufficient, and there would be only fixed names like "controller" or "agent".

Reply:
We don't need to change this right now. But I think from an API perspective, it is a bit weird to have "create" semantics and be restricted in the name you can use.

Regarding multiple clients, I think it will only matter if multiple users request bundles at the same time and if we introduce parameters to the bundle collection requests. For example, something that Cody suggested recently is the ability to restrict the log messages collected to a certain time frame (e.g. within the last 2 hours).

Reply:
Yes, filters are quite useful, but we need to clean up legacy logs first.
Thus I would like to implement filters in future.